src/refract/RenderFormURLEncodedVisitor.cc

```diff
@@ -0,0 +1,198 @@
+//
+//  RenderFormURLEncodedVisitor.cc
+//  librefract
+//
+//  Created by Kyle Fuller on 17/09/16.
+//  Copyright (c) 2016 Apiary Inc. All rights reserved.
+//
+
+#include <iomanip>
+#include "VisitorUtils.h"
+
+#include "RenderFormURLEncodedVisitor.h"
+
+namespace refract
+{
+
+    namespace {
+        template <typename T>
+        void IncludeMembers(const T& element, typename T::ValueType& members)
+        {
+            const typename T::ValueType* value = GetValue<T>(element);
+
+            if (!value) {
+                return;
+            }
+
+            for (typename T::ValueType::const_iterator it = value->begin();
+                 it != value->end();
+                 ++it) {
+
+                if (!(*it) || (*it)->empty()) {
+                    continue;
+                }
+
+                if ((*it)->element() == "ref") {
+                    HandleRefWhenFetchingMembers<T>(*it, members, IncludeMembers<T>);
+                    continue;
+                }
+
+                members.push_back(*it);
+            }
+        }
+    }
+
+    RenderFormURLEncodedVisitor::RenderFormURLEncodedVisitor(const std::string &key) : baseKey(key) {
+    }
+
+    RenderFormURLEncodedVisitor::~RenderFormURLEncodedVisitor() {
+    }
+
+    void RenderFormURLEncodedVisitor::operator()(const IElement& element) {
+        VisitBy(element, *this);
+    }
+
+    void RenderFormURLEncodedVisitor::operator()(const MemberElement& element) {
+        if (!element.value.first) {
+            return;
+        }
+
+        StringElement *keyElement = static_cast<StringElement*>(element.value.first);
```

Comment:
This needs to be updated to handle the case where there isn't a `StringElement` but instead a `SelectElement`.

```apib
# GET /

+ Response 200 (application/x-www-form-urlencoded)
    + Attributes
        + One Of
            + foo
            + bar
```